auto_joiner.py

```diff
@@ -169,6 +169,15 @@ def wait_until_found(sel, timeout):
         print("Timeout waiting for element.")
         return None
 
+def wait_until_found_xpath(sel, timeout):
+    try:
+        element_present = EC.visibility_of_element_located((By.XPATH, sel))
+        WebDriverWait(browser, timeout).until(element_present)
+
+        return browser.find_element_by_xpath(sel)
+    except exceptions.TimeoutException:
+        print("Timeout waiting for element.")
+        return None
```

Comment:
This is just not needed, please delete all of this and use the wait_until_found function